app/models/miq_storage_metric.rb

```diff
@@ -263,8 +257,7 @@ def self.derived_metrics_class_name
   # Called from subclass of MiqStorageMetric.
   #
   def self.derived_metrics_class
-    return nil if derived_metrics_class_name.nil?
-    derived_metrics_class_name.constantize
+    derived_metrics_class_name.try(:constantize)
```

Comment:
Is it possible to use [try!](http://apidock.com/rails/v4.2.1/Object/try%21) here instead? Its semantics is a little bit closer to the semantics of the original method body, and it is also a little bit faster than `try` (when derived_metrics_class_name is not nil; when derived_metrics_class_name is nil then the performance is the same).

```ruby
type = nil

Benchmark.ips do |x|
  puts "type = #{type.inspect}"
  x.report('try') { type.try(:constantize) }
  x.report('try!') { type.try!(:constantize) }
  x.compare!
end

type = "String"

Benchmark.ips do |x|
  puts "type = #{type.inspect}"
  x.report('try') { type.try(:constantize) }
  x.report('try!') { type.try!(:constantize) }
  x.compare!
end
```

```
type = nil
Calculating -------------------------------------
                 try    48.145k i/100ms
                try!    48.401k i/100ms
-------------------------------------------------
                 try      1.860M (± 2.5%) i/s -      9.292M
                try!      1.836M (± 5.1%) i/s -      9.148M

Comparison:
                 try:  1859691.1 i/s
                try!:  1836304.1 i/s - 1.01x slower


type = "String"
Calculating -------------------------------------
                 try    11.687k i/100ms
                try!    13.363k i/100ms
-------------------------------------------------
                 try    198.986k (± 2.8%) i/s -      1.005M
                try!    236.428k (± 5.8%) i/s -      1.189M

Comparison:
                try!:   236428.4 i/s
                 try:   198986.1 i/s - 1.19x slower
```